### ISSUE TITLE
Dropout may need to be added to the code

### ISSUE THREAD

**Issue by jijunhao** [lucidrains/denoising-diffusion-pytorch#202]
Can dropout be added to the network, such as inside the attention layer, some linear layer? 

For example

```
class Attention(nn.Module):
    def __init__(self, dim, heads = 4, dim_head = 32, dropout = 0.):
        super().__init__()
        hidden_dim = dim_head * heads

        self.scale = dim_head ** -0.5
        self.heads = heads

        self.attend = nn.Softmax(dim = -1)
        self.dropout = nn.Dropout(dropout)

        self.to_qkv = nn.Conv2d(dim, hidden_dim * 3, 1, bias = False)
        self.to_out = nn.Conv2d(hidden_dim, dim, 1)

    def forward(self, x):
        b, c, h, w = x.shape

        qkv = self.to_qkv(x).chunk(3, dim = 1)
        q, k, v = map(lambda t: rearrange(t, 'b (h c) x y -> b h c (x y)', h = self.heads), qkv)

        q = q * self.scale

        # softmax(Q*K^T/sqrt(dim_head))*V
        sim = einsum('b h d i, b h d j -> b h i j', q, k)
        attn = self.attend(sim)
        attn = self.dropout(attn)
        out = einsum('b h i j, b h d j -> b h i d', attn, v)
        out = rearrange(out, 'b h (x y) d -> b (h d) x y', x = h, y = w)
        return self.to_out(out)
```

**Comment by jijunhao** [comment]
Oh, I see it in simple-diffusion, Sorry!
